internal/schemautil/service.go

```diff
@@ -636,12 +636,17 @@ func copyServicePropertiesFromAPIResponseToTerraform(
 		}
 	}
 
-	userConfig, err := apiconvert.FromAPI(userconfig.ServiceTypes, serviceType, s.UserConfig)
+	newUserConfig, err := apiconvert.FromAPI(userconfig.ServiceTypes, serviceType, s.UserConfig)
 	if err != nil {
 		return err
 	}
 
-	if err := d.Set(serviceType+"_user_config", NormalizeIpFilter(d.Get(serviceType+"_user_config"), userConfig)); err != nil {
+	// Mutates user config inplace
+	oldUserConfig := d.Get(serviceType + "_user_config")
```

Comment:
could `oldUserConfig` please be renamed to `curUserConfig` or similar, since at this point in time in the code, the user config was not changed yet, thus it is not an old one, yet

Reply:
TF uses naming old and new for diff suppress func.